src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestCreationParameters.cs

```diff
@@ -0,0 +1,129 @@
+﻿//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.IdentityModel.Protocols.SignedHttpRequest
+{
+    /// <summary>
+    /// A delegate that will be called to create and add additional claim(s), if set.
+    /// </summary>
+    /// <param name="payload">A SignedHttpRequest payload.</param>
+    /// <param name="signedHttpRequestDescriptor">A structure for wrapping parameters needed for SignedHttpRequest creation.</param>
+    public delegate void AdditionalClaimCreator(IDictionary<string, object> payload, SignedHttpRequestDescriptor signedHttpRequestDescriptor);
```

Comment:
Can we add a property on SignedHttpRequestDescriptor?
Do we need this delegate?

Reply:
It can be a property. I had a delegate counterpart on the validation side so I wanted a delegate on the creation side, for consistency.
Should I remove the `AdditionalClaimValidatorAsync` delegate?
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/a7c7b608fbd0da084e479c531b45120cf8f738c6/src/Microsoft.IdentityModel.Protocols.SignedHttpRequest/SignedHttpRequestValidationParameters.cs#L39-L46

Reply:
Addressed with https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1260/commits/c465dc691d052b7ad13baee9219b69eb972c65d0 and https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1260/commits/c5694079f20b5961bf3b4f100c12d9a144a8f79f